File: Client/Bridge.ts
import axios, { AxiosResponse } from 'axios'
import qs from 'qs'

// const BASE = 'http://34.94.109.123'
const BASE = 'http://localhost:3000'

export class Bridge {
  jwt: string
  authenticated = true


  constructor(jwt?: string) {
    if (!jwt) this.authenticated = false
    else this.jwt = jwt
  }

  post(url: string, bodyObject: any): Promise<AxiosResponse<any>> {

    let body = (qs.stringify(bodyObject))

    let config = {
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded'
      }
    }

    if (this.authenticated) config.headers['x-access-token'] = this.jwt

    return axios.post(BASE + url, body, config)
  }

  get(url: string): Promise<AxiosResponse<any>> {

    let config = { headers: {} }

    if (this.authenticated) config.headers['x-access-token'] = this.jwt

    return axios.get(BASE + url, config)
  }

}

File: test_boogie/lib/Bridge.dart
import 'package:http/http.dart' as http;


class Bridge {
  String jWT;


  // static const String BASE = ''
  // static const String BASE = 'http://10.245.81.79:3000';
  static const String BASE = 'http://localhost:3000';

  Bridge ([this.jWT]);

  dynamic getHeaders () {
    if(jWT != null) return {
      'X-access-token': jWT
    };
    else return null;
  }

  Future<http.Response> post (String url, Map<String, String> data) {
    return  http.post(BASE + url,
    headers: getHeaders(),
    body: data);
  }

  Future<http.Response> httpGet(String url) {
    return http.get(BASE + url, headers: getHeaders());
  }
}